lib/oxide-vpc/src/engine/router.rs

```diff
@@ -116,54 +117,60 @@ impl fmt::Display for RouterTargetInternal {
 }
 
 // Return the priority for a given IP subnet. The priority is based on
-// the subnet's prefix length. Specifically, it is given the following
-// value.
+// the subnet's prefix length and the type of router the rule belongs to.
+// Specifically, it is given the following value:
 //
 // ```
-// priroity = max_prefix_len - prefix len + 10
+// priority = (((max_prefix_len - prefix len) << 1) | is_system) + 10
 // ```
 //
 // `max_prefix_len` is the maximum prefix length for a given IP
 // CIDR type: `32` for IPv4, `128` for IPv6.
 //
 // `prefix_len` comes from the passed in `cidr` argument.
 //
+// One bit is used to ensure that 'custom' router rules take precedence
```

Comment:
Q: is there a test that specifically tests precedence? It's probably already captured, but I thought to ask.

Reply:
Thanks for taking a look. The new system vs. custom route precedence is tested in the new `intra_subnet_routes_with_custom` test on an identical destination CIDR: https://github.com/oxidecomputer/opte/blob/789b927aed87d251b5e318431f373815f8f7dfc2/lib/oxide-vpc/tests/integration_tests.rs#L4064-L4090

I don't believe we have any explicit tests on e.g. the priority values we're using to get LPM-like behaviour, though.